oak-core/src/test/java/org/apache/jackrabbit/oak/query/UnionQueryTest.java

```diff
@@ -74,6 +77,52 @@ public void after() throws Exception {
         root.commit();
     }
 
+    @Test
+    public void testMergeSortedVsConcat() throws  Exception {
```

Comment:
While this is a good test in general, and we should anyway keep it... Does this new test fail if the changes in the UnionQueryImpl above are _not_ done, that is, with the old code? (I didn't test, but I think the test doesn't fail...).

I think it would be possible to add a test that fails with the old code in UnionQueryImpl, but I guess I need to see myself (using a custom Query subclass in the test).